File: app/src/main/java/com/june/mvpkj/j/base/BasePresenter.java
package com.june.mvpkj.j.base;

/**
 * @CreateTime: 2022/1/14
 * @author: jjzhu
 */
public class BasePresenter<V extends BaseView> {
    public V view;

    public BasePresenter(V view) {
        this.view = view;
    }

    public void detachView(){
        this.view = null;
    }

    public boolean isViewAttached(){
        return this.view != null;
    }
}

File: app/src/main/java/com/june/mvpkj/k/base/BasePresenter.kt
package com.june.mvpkj.k.base

/**
 *@CreateTime: 2022/1/14
 *@author: jjzhu
 */
open class BasePresenter<V : BaseView>(protected var view: V?) {

    fun detachView(){
        this.view = null
    }

    fun isViewAttached(): Boolean{
        return this.view != null
    }
}